src/custom-element.js

```diff
@@ -987,9 +977,17 @@ function createBaseCustomElementClass(win) {
       }
       if (!this.ampdoc_) {
         // Ampdoc can now be initialized.
-        const ampdocService = Services.ampdocServiceFor(
-            this.ownerDocument.defaultView);
-        this.ampdoc_ = ampdocService.getAmpDoc(this);
+        const win = this.ownerDocument.defaultView;
+        const ampdocService = Services.ampdocServiceFor(win);
+        const ampdoc = ampdocService.getAmpDoc(this);
+        this.ampdoc_ = ampdoc;
+        // Load the pre-stubbed extension if needed.
+        const extensionId = this.tagName.toLowerCase();
+        if (isStub(this.implementation_) &&
+            !ampdoc.declaresExtension(extensionId)) {
+          Services.extensionsFor(win).installExtensionForDoc(
```

Comment:
Just leaving this question here: If we only run the side effect when the first element instantiates, what do we do for extensions that don't have an element?